From 389b2ffb8c65b146c32b83719ff5cf9aa1faa340 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Sun, 25 Nov 2018 21:39:37 +0100
Subject: [PATCH 0220/1239] i2c: mv_i2c: implement handler required by "i2c
 reset" cmd

After this change the "i2c reset" command can be used which will allow
to re-initialize I2C Controller.

Change-Id: I39703611bcf8d4bfe03ef619e5dd5df8a9d59efb
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
---
 drivers/i2c/mv_i2c.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/drivers/i2c/mv_i2c.c b/drivers/i2c/mv_i2c.c
index 9bde9cd599..d1803a6046 100644
--- a/drivers/i2c/mv_i2c.c
+++ b/drivers/i2c/mv_i2c.c
@@ -585,9 +585,18 @@ static int mv_i2c_probe(struct udevice *bus)
 	return 0;
 }
 
+static int mv_i2c_reset(struct udevice *bus)
+{
+	struct mv_i2c_priv *priv = dev_get_priv(bus);
+
+	i2c_reset(priv->base);
+	return 0;
+}
+
 static const struct dm_i2c_ops mv_i2c_ops = {
 	.xfer		= mv_i2c_xfer,
 	.set_bus_speed	= mv_i2c_set_bus_speed,
+	.deblock	= mv_i2c_reset,
 };
 
 static const struct udevice_id mv_i2c_ids[] = {
-- 
2.29.0

